### ISSUE TITLE
twisted.mail.test.test_mail.ProcessAliasTestCase leaves the reactor unclean

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2266]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#2266|
|Type|defect|
|Created|2006-12-06 02:37:42Z|

<details><summary>Searchable metadata</summary>

```
trac-id__2266 2266
type__defect defect
reporter__jml jml
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__tests tests
time__1165372662000000 1165372662000000
changetime__1193952005000000 1193952005000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

I think this has been solved in [19640]. Do you confirm ?

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

Forget it, it seems it's an intermittent problem.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve commented|
|-|-|

(In [21458]) Use a MockProcessAlias that actually doesn't launch a process.

Refs [#2266](https://github.com/twisted/twisted/issues/2266)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve removed owner|
|-|-|

Ready to review in mail-alias-unclean-2266.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve commented|
|-|-|

Merged forward in mail-alias-unclean-2266-2.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set owner to @therve|
|-|-|


  * test_processAlias misspells _transferred_.
  * I'd say something more specific in the docstring for test_aliasResolution, maybe mention the types of aliases and what each corresponds to.
  * twisted.mail.alias imports twisted.mail.smtp imports twisted.internet.reactor at module scope, so there's no reason for twisted.mail.alias.ProcessAlias.spawnProcess to nest the reactor import instead of having it at module scope

Hmmm.  This gets rid of the unclean errors from this test.  It no longer exercises the process spawning code at all though.  It wasn't really _testing_ that behavior before though.  Probably there should be a ticket for adding test coverage for ```ProcessAlias```.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve commented|
|-|-|

(In [21526]) Process review comments

Refs [#2266](https://github.com/twisted/twisted/issues/2266)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_10"></a>|@therve removed owner|
|-|-|

Thanks for the comments, the docstring of test_aliasResolution was a bit hard to produce, I hope it will be fine enough.

Note that test_processAlias does spawn a real process, and check its result. But I think file wrapper are untested.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set owner to @therve|
|-|-|

Cool, merge please :)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_12"></a>|@therve set status to `closed`|
|-|-|

(In [21528]) Merge mail-alias-unclean-2266-2

Author: therve
Reviewer: exarkun
Fixes [#2266](https://github.com/twisted/twisted/issues/2266)

Remove unclean errors from twisted.mail.test.test_mail.ProcessAliasTestCase.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
